### PR TITLE
`Number` rule improvements

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #531, #536

### CODE REVIEW

**Review by cebe — APPROVED** [review]

**Comment by arogachev** [comment]
@cebe Thanks for the review, but PR was at the draft stage. Now it's ready for review.

File: src/Rule/Integer.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Rule;
+
+use Attribute;
+use Closure;
+use Yiisoft\Validator\WhenInterface;
+
+/**
+ * Defines validation options to check that the value is a integer number.
```
Comment (samdark) [inline src/Rule/Integer.php:12]:
```suggestion
 * Defines validation options to check that the value is an integer number.
```

File: src/Rule/AbstractNumber.php [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+use Yiisoft\Validator\SkipOnEmptyInterface;
+use Yiisoft\Validator\SkipOnErrorInterface;
+use Yiisoft\Validator\WhenInterface;
+
+/**
+ * Defines validation options to check that the value is a number.
+ *
+ * The format of the number must match the regular expression specified in {@see AbstractNumber::$pattern}. Optionally,
+ * you may configure the {@see AbstractNumber::$min} and {@see AbstractNumber::$max} to ensure the number is within
+ * certain range.
```
Comment (samdark) [inline src/Rule/AbstractNumber.php:21]:
```suggestion
 * a certain range.
```

File: src/Rule/Integer.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ *
+ * The format of the number must match the regular expression specified in {@see Integer::$pattern}. Optionally, you may
+ * configure the {@see Integer::$min} and {@see Integer::$max} to ensure the number is within certain range.
```
Comment (samdark) [inline src/Rule/Integer.php:15]:
```suggestion
 * configure the {@see Integer::$min} and {@see Integer::$max} to ensure the number is within a certain range.
```

File: src/Rule/Number.php [hunk 4]
```diff
@@ -6,226 +6,90 @@
...
 use Yiisoft\Validator\WhenInterface;
 
 /**
  * Defines validation options to check that the value is a number.
  *
- * The format of the number must match the regular expression specified in {@see Number::$integerPattern}
- * or {@see Number::$numberPattern}. Optionally, you may configure the {@see Number::min()} and {@see Number::max()}
- * to ensure the number is within certain range.
+ * The format of the number must match the regular expression specified in {@see Number::$pattern}. Optionally, you may
+ * configure the {@see Number::$min} and {@see Number::$max} to ensure the number is within certain range.
```
Comment (samdark) [inline src/Rule/Number.php:15]:
```suggestion
 * configure the {@see Number::$min} and {@see Number::$max} to ensure the number is within a certain range.
```

File: tests/Benchmark/MainBench.php [hunk 5]
```diff
@@ -10,8 +10,8 @@
 use PhpBench\Benchmark\Metadata\Annotations\ParamProviders;
 use PhpBench\Benchmark\Metadata\Annotations\Revs;
 use PhpBench\Benchmark\Metadata\Annotations\Warmup;
+use phpDocumentor\Reflection\Types\Integer;
```
Comment (samdark) [inline tests/Benchmark/MainBench.php:13]: Does not look right.

Comment (samdark) [reply tests/Benchmark/MainBench.php:13]:
```suggestion
use Yiisoft\Validator\Rule\Integer;
```

**Review by samdark — APPROVED** [review]
All 👍 except minor stuff I've mentioned in comments.

**Pull request merged**